tests/python/pytorch/graphbolt/impl/test_sampled_subgraph_impl.py

```diff
@@ -0,0 +1,121 @@
+import pytest
+import torch
+from dgl.graphbolt.impl.sampled_subgraph_impl import (
+    exclude_edges,
+    SampledSubgraphImpl,
+)
+
+
+def _assert_container_equal(lhs, rhs):
+    if isinstance(lhs, torch.Tensor):
+        assert isinstance(rhs, torch.Tensor)
+        assert torch.equal(lhs, rhs)
+    elif isinstance(lhs, tuple):
+        assert isinstance(rhs, tuple)
+        assert len(lhs) == len(rhs)
+        for l, r in zip(lhs, rhs):
+            _assert_container_equal(l, r)
+    elif isinstance(lhs, dict):
+        assert isinstance(rhs, dict)
+        assert len(lhs) == len(rhs)
+        for key, value in lhs.items():
+            assert key in rhs
+            _assert_container_equal(value, rhs[key])
+
+
+@pytest.mark.parametrize("is_reverse", [True, False])
+def test_exclude_edges_homo(is_reverse):
+    node_pairs = (torch.tensor([0, 2, 3]), torch.tensor([1, 4, 2]))
+    if is_reverse:
+        reverse_row_node_ids = torch.tensor([10, 15, 11, 24, 9])
+        reverse_column_node_ids = torch.tensor([10, 15, 11, 24, 9])
```

Comment:
@peizhou001 do we compact row and column respectively or altogether?

Reply:
We compact them together. But according to our previous discussion, row and column may not be both compressed in some algorithm.

Reply:
My implementation does not assume they are compacted together. I will add tests on the cases of single compaction.